lib/Zonemaster/Backend/DB.pm

```diff
@@ -316,6 +340,37 @@ sub process_unfinished_tests {
     }
 }
 
+sub select_unfinished_tests {
+    my ( $self, $queue_label, $test_run_timeout ) = @_;
+
+    if ( $queue_label ) {
+        my $sth = $self->dbh->prepare( "
+            SELECT hash_id, results
+            FROM test_results
+            WHERE test_start_time < ?
+            AND progress > 0
+            AND progress < 100
+            AND queue = ?" );
+        $sth->execute(    #
+            $self->format_time( time() - $test_run_timeout ),
+            $queue_label,
+        );
+        return $sth;
```

Comment:
Have you considered having only one `return` statement in this function?

Reply:
Yeah, I really think this pattern:
```perl
if ( ... ) {
    return ...;
} else {
    return ...;
}
```
is better than this pattern:
```perl
my $z;
if ( ... ) {
    $z = ...;
} else {
    $z = ...;
}
return $z;
```

Reply:
I guess it could be useful to consider a third pattern here as well:
```perl
...
if ( ... ) {
    return ...;
}
...
return ...;
```
To me this pattern is problematic because it's easy misunderstand the structure of the control flow if you don't spot the return statement in the middle of things. But I don't think the problem is the number of `return` statements.

It's different when **all** branches of an `if / elsif / else` construct return (or throw). In this case the `if / elsif / else` construct is the last thing that happens in the procedure, so there's no code after it where you can get confused about the control flow structure.